samples/drivers/gpio/custom_dts_binding/README.rst

```diff
@@ -0,0 +1,62 @@
+.. _gpio-custom-dts-binding-sample:
+
+GPIO with custom Devicetree binding
+###################################
+
+Overview
+********
+
+In Zephyr, all hardware-specific configuration is described in the devicetree.
+
+Consequently, also GPIO pins are configured in the devicetree and assigned to
+a specific purpose using a compatible.
+
+This is in contrast to other embedded environments like Arduino, where e.g.
+the direction (input / output) of a GPIO pin is configured in the application
+firmware.
+
+For typical use cases like LEDs or buttons, the existing `gpio-leds` or
+`gpio-keys` compatibles can be used.
+
+This sample demonstrates how to use a GPIO pin for other purposes with a
+custom dts binding.
+
+We assume that a load with high current demands should be switched on or off
+via a MOSFET. The custom DTS binding for the power output controlled via a
+GPIO pin is specified in the file `dts/bindings/power-output.yaml`. The gate
+driver for the MOSFET would be connected to the pin as specified in the
+`.overlay` file in the boards folder.
+
+Building and Running
+********************
+
+For each board that should be supported, a `.overlay` file has to be defined
+in the `boards` subfolder.
+
+Building and Running for ST Nucleo L073RZ
+=========================================
+The sample can be built and executed for the
+:ref:`nucleo_l073rz_board` as follows:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/drivers/gpio/custom_dts_binding
+   :board: nucleo_l073rz
+   :goals: build flash
+   :compact:
+
+For demonstration purposes, we use the GPIO pin of the built-in LED.
```

Comment:
I think this sample can be improved by _not_ using the LED GPIO pin, as using the LED GPIO pin leaves out one important step: Having to setup the pinmux.

Reply:
This statement depends on platform. On STM32, this is not needed (only required for PMW LED).
But I agree, ideally this sample should not require additional change.

Reply:
Ah, ok, was not aware that the pinmux setup is necessary for other platforms. Can you suggest a board that requires pinmux setup and that one of you has available for testing? I can add an additional overlay for that. (assuming those other platforms handle pinmux via dts and not with the `pinmux.c` file as for STM32)

Reply:
> I think this sample can be improved by _not_ using the LED GPIO pin, as using the LED GPIO pin leaves out one important step: Having to setup the pinmux.

Setting pinmux is currently only required on NXP platform and hopefully this will soon change. As part of the work on supporting pinmux configuration in DTS we want the driver to be responsible for configuring the pins. The same approach is used by Linux.

I propose to leave this part as is.